### DISCUSSION TITLE
Use FastAPI's background tasks as a dependency

### DISCUSSION THREAD

**Discussion by kirantyefun** [fastapi-users/fastapi-users#1306] [Q&A]
My usecase is: I want to send a confirmation email right after user registers.

FastAPI has its own version of BackgroundTasks for simple tasks such as sending an email which would not block any response. But the caveat is I can only use Background Task as either a parameter in path operation, or in a dependency or in a sub-dependency. 
I could not find the right place to use BackgroundTask. Please suggest.

**Comment by frankie567** [comment]
See: https://github.com/fastapi-users/fastapi-users/pull/1288#issuecomment-1733027809
